### DISCUSSION TITLE
Make docker release images multi-platform (arm + amd)

### DISCUSSION THREAD

**Discussion by mistercrunch** [apache/superset#27043] [Ideas]
As it came up in https://github.com/apache/superset/issues/26783, it'd be nice to have our official release docker images be multi-platform (support both amd and arm). To be clear, we currently support both platforms, but not under the same tag, meaning currently we have say `lean` and `lean-arm`, but in thoery, `lean` could contain both `arm` and `amd` for convenience. From my understanding that would also make downstream constructs like docker-compose and helm more universal/simpler

Copy pasting the most-relevant comment from there here:

Two approaches:
- I think there's a way to merge the manifests post build, but that seemed tricky/annoying
- pass two platform arguments to docker build (which doubles the build time) and we only want on-release

I'd go with option 2, and on-release only, but it'd require doing a bit of work around the build script. Overall the process would be:

- add support for multi-platform argument [here](https://github.com/apache/superset/blob/master/scripts/build_docker.py#L206-L209)
- remove platform from the matrix [here](https://github.com/apache/superset/blob/master/.github/workflows/docker-release.yml#L46)
- pass the two platforms in the call [here](https://github.com/apache/superset/blob/master/.github/workflows/docker-release.yml#L95)
- re-trigger the job for old releases (easy)

**Comment by arundh93** [comment]
@mistercrunch One suggestion for helm installation. I think we should update the docs to mention that we might have to override https://github.com/apache/superset/blob/master/helm/superset/values.yaml#L189 for helm installation on a ARM platform.

This is needed because users try to install the superset helm chart will face issue since the `dockerise` tag in the image default is not for ARM platform.

Let me know if I can help in updating the docs for the same or if you have any other thoughts

**Reply by mistercrunch** [reply]
Ok feel free to update the docs to reflect that. I don't use helm / docker personally so I'm not the best person to document or maintain that. The main reason I worked on docker over the past few weeks was because it was slowing down CI (was taking 50 minutes to run CI for each commit in a PR).

**Reply by mistercrunch** [reply]
Also I think that if we implement multi-platform images as proposed in this issue then helm should just work without messing with values

**Reply by arundh93** [reply]
Sure let's wait for the multi platform images to go in first then.

**Reply by mistercrunch** [reply]
Yup, let's wait, maybe someone will pick this up. 

Leaves me wondering who uses helm on arm, like do people run k8s on their Mac? Run superset on arm-powered EC2 boxes? Why?

**Comment by mistercrunch** [comment]
Here, this should help https://github.com/apache/superset/pull/27055

**Comment by mhassanzahid** [comment]
Hello,

Just a question regarding multi-arch images, we are using superset version `4.0.2` (which is multi-arch)  as a base image and installing 4 dependencies for using the "ENABLE FORECAST" feature i.e. 

- lunarcalendar
- tqdm
- pystan
- prophet

The problem we are facing is, httpstan part of pystan doesn't have an arm64 binary (atleast according to what i see on pypi, i can be wrong) and our docker build is failing because of this for arm64 images, is this resolvable with any alternatives, or is it not possible to build for arm64 architecture with "ENABLE FORECAST" enabled?

**Comment by mistercrunch** [comment]
Seems like more of a pystan problem than a Superset one. Are you using arm64 as a production deploy? For development (if you're actively working on this feature) and as a workaround you could fire up an EC2 amd box of some kind.

**Comment by mhassanzahid** [comment]
Yes, we're using arm64 nodes as production deploy. Is there an alternative to pystan then in this case, or any ways we could get around this issue?

**Reply by mistercrunch** [reply]
I'd recommend opening an issue here https://github.com/stan-dev/pystan

Also you may want to use amd64 architecture in production to avoid this whole family of issues. Many of us are on OSX for local dev, but I'd assume 95%+ of the community uses amd for production

**Comment by mistercrunch** [comment]
Closing this discussion as the main goal has been largely achieved. We can re-open specific issues/discussions that are more targeted from this point on.
